conan/api/subapi/config.py

```diff
@@ -229,8 +235,9 @@ def set_core_confs(self, core_confs):
         confs.loads("\n".join(core_confs))
         confs.validate()
         self._cli_core_confs = confs
-        # Last but not least, apply the new configuration
-        self.conan_api.reinit()
+        with interprocess_write_lock(self.conan_api):
```

Comment:
This happens at every ``parse_args`` which is every command. 
Are we sure that we need to lock here? This is not really changing the files in the cache, why it is a write lock?

Reply:
Reverted, commit 52641b1de65341e283cfc557542e619dbcf01a0a